qiskit/circuit/quantumcircuit.py

```diff
@@ -369,21 +370,52 @@ def compose(self, other, qubits=None, clbits=None, front=False, inplace=False):
         Raises:
             CircuitError: if composing on the front.
             QiskitError: if ``other`` is wider or there are duplicate edge mappings.
+
+        Examples:
+
+            >>> lhs.compose(rhs, qubits=[3, 2], inplace=True)
+
+                        ┌───┐                   ┌─────┐                ┌───┐
+            lqr_1_0: ───┤ H ├───    rqr_0: ──■──┤ Tdg ├    lqr_1_0: ───┤ H ├───────────────
+                        ├───┤              ┌─┴─┐└─────┘                ├───┤
+            lqr_1_1: ───┤ X ├───    rqr_1: ┤ X ├───────    lqr_1_1: ───┤ X ├───────────────
+                     ┌──┴───┴──┐           └───┘                    ┌──┴───┴──┐┌───┐
+            lqr_1_2: ┤ U1(0.1) ├  +                     =  lqr_1_2: ┤ U1(0.1) ├┤ X ├───────
+                     └─────────┘                                    └─────────┘└─┬─┘┌─────┐
+            lqr_2_0: ─────■─────                           lqr_2_0: ─────■───────■──┤ Tdg ├
+                        ┌─┴─┐                                          ┌─┴─┐        └─────┘
+            lqr_2_1: ───┤ X ├───                           lqr_2_1: ───┤ X ├───────────────
+                        └───┘                                          └───┘
+            lcr_0: 0 ═══════════                           lcr_0: 0 ═══════════════════════
+                                                                                           
+            lcr_1: 0 ═══════════                           lcr_1: 0 ═══════════════════════
+
         """
-        from qiskit.converters.circuit_to_dag import circuit_to_dag
-        from qiskit.converters.dag_to_circuit import dag_to_circuit
         if front:
             raise CircuitError("Front composition of QuantumCircuit not supported yet.")
 
-        dag_self = circuit_to_dag(self)
-        dag_other = circuit_to_dag(other)
-        dag_self.compose(dag_other, qubits=qubits, clbits=clbits, front=front)
-        composed_circuit = dag_to_circuit(dag_self)
-        if inplace:  # FIXME: this is just a hack for inplace to work. Still copies.
-            self.__dict__.update(composed_circuit.__dict__)
-            return None
-        else:
-            return dag_to_circuit(dag_self)
+        if isinstance(other, QuantumCircuit):
+            from qiskit.converters.circuit_to_dag import circuit_to_dag
+            from qiskit.converters.dag_to_circuit import dag_to_circuit
+
+            dag_self = circuit_to_dag(self)
+            dag_other = circuit_to_dag(other)
+            dag_self.compose(dag_other, qubits=qubits, clbits=clbits, front=front)
+            composed_circuit = dag_to_circuit(dag_self)
+            if inplace:  # FIXME: this is just a hack for inplace to work. Still copies.
+                self.__dict__.update(composed_circuit.__dict__)
+                return None
+            else:
+                return dag_to_circuit(dag_self)
```

Comment:
why do you need the dag_to_circuit(dag_self) cant it just be composed_circuit

Reply:
good catch! fixed.